instrumentation-api-annotation-support/src/main/java/io/opentelemetry/instrumentation/api/annotation/support/MethodSpanAttributesExtractor.java

```diff
@@ -54,18 +59,81 @@ protected void onStart(AttributesBuilder attributes, REQUEST request) {
   protected void onEnd(
       AttributesBuilder attributes, REQUEST request, @Nullable RESPONSE response) {}
 
-  private static class MethodSpanAttributeBinder extends BaseAttributeBinder {
-    private final ParameterAttributeNamesExtractor parameterAttributeNamesExtractor;
+  /**
+   * Creates a binding of the parameters of the traced method to span attributes.
+   *
+   * @param method the traced method
+   * @return the bindings of the parameters
+   */
+  private AttributeBindings bind(Method method) {
+    AttributeBindings bindings = EmptyAttributeBindings.INSTANCE;
+
+    Parameter[] parameters = method.getParameters();
+    if (parameters == null || parameters.length == 0) {
+      return bindings;
+    }
+
+    String[] attributeNames = parameterAttributeNamesExtractor.extract(method, parameters);
+    if (attributeNames == null || attributeNames.length != parameters.length) {
```

Comment:
Can attributeNames be null? There are a few more null checks, if you can remove unnecessary ones it's good - we avoid them to make the code easier to read for someone that isn't familiar with it.

Reply:
The definition of the functional interface allows for it, `null` indicating that the method has no parameters to map to attributes:  https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api-annotation-support/src/main/java/io/opentelemetry/instrumentation/api/annotation/support/ParameterAttributeNamesExtractor.java#L26

I could change that to specify returning an empty array if preferred.

Reply:
confusingly, I think `@Nullable String[]` means a non-null array of nullable strings, while `String @Nullable []` means a nullable array of non-null strings

Reply:
That's terrifying.  I do defensively check for either the array being null or each element being null, so would that be `@Nullable String @Nullable []`?

Reply:
I'll review the two implementations to see if it's possible that either will return `null`, if not I can change the annotation.  Is it still worth having the defensive check even if the interface states that it shouldn't be `null`?  I hate the idea of a bug in instrumentation leading to errors at runtime.

Reply:
I think @anuraaga's point is that ParameterAttributeNamesExtractor.extract() returns `@Nullable String[]`, so you don't need to check if the array is `null`

Reply:
Got it, so if the array itself can't be null (and that does seem to be the case) then the null check is unnecessary and I can leave the annotations as they are.  Thanks!

Reply:
we are (slowly) rolling out [NullAway](https://github.com/uber/NullAway) checking, which will hopefully help (or maybe add to?) some of the confusion

Reply:
(also, I could be wrong about how NullAway interprets `@Nullable String[]`, my assumption is based on how Checker Framework works, it will be good to get confirmation here from @anuraaga)

Reply:
I chatted with @anuraaga, NullAway (luckily) does not annotate the contents of the array (like the more powerful, but infinitely more confusing Checker Framework), so `@Nullable String[]` means what you thought it meant (Nullable applies to the array, not to the strings). Sorry for the confusion.